photutils/detection/tests/test_findstars.py

```diff
@@ -94,23 +102,28 @@ def test_irafstarfind(self, threshold, fwhm):
 
     def test_irafstarfind_nosources(self):
         data = np.ones((3, 3))
-        t = irafstarfind(data, threshold=10, fwhm=1)
+        starfinder = IRAFStarFinder(threshold=10, fwhm=1)
+        t = starfinder(data) 
         assert len(t) == 0
 
     def test_irafstarfind_sharpness(self):
         """Sources found, but none pass the sharpness criteria."""
-        t = irafstarfind(DATA, threshold=50, fwhm=1.0, sharplo=2.)
+        starfinder = IRAFStarFinder(threshold=50, fwhm=1.0, sharplo=2.)
+        t = starfinder(DATA)
         assert len(t) == 0
 
     def test_irafstarfind_roundness(self):
         """Sources found, but none pass the roundness criteria."""
-        t = irafstarfind(DATA, threshold=50, fwhm=1.0, roundlo=1.)
+        starfinder = IRAFStarFinder(threshold=50, fwhm=1.0, roundlo=1.)
+        t = starfinder(DATA)
         assert len(t) == 0
 
     def test_irafstarfind_sky(self):
-        t = irafstarfind(DATA, threshold=25.0, fwhm=2.0, sky=10.)
+        starfinder = IRAFStarFinder(threshold=25.0, fwhm=2.0, sky=10.)
+        t = starfinder(DATA)
         assert len(t) == 4
 
     def test_irafstarfind_largesky(self):
-        t = irafstarfind(DATA, threshold=25.0, fwhm=2.0, sky=100.)
+        starfinder = IRAFStarFinder(threshold=25.0, fwhm=2.0, sky=100.)
+        t = starfinder(DATA)
         assert len(t) == 0
```

Comment:
I think we tend to keep the tests for deprecated but not yet removed functionality in astropy core and affiliated packages I looked into, so +1 keeping it and adding new ones for the new API.